src/librustc/ty/maps/on_disk_cache.rs

```diff
@@ -604,6 +610,7 @@ impl<'a, 'tcx, 'x> SpecializedDecoder<interpret::AllocId> for CacheDecoder<'a, '
                 this.with_position(shorthand, |this| interpret::AllocId::decode(this))
             }
         )?;
+        // FIXME: Looks like overwrites aren't allowed here
```

Comment:
We could asseet that the result is the same. That would be fine I guess.

Reply:
That won't work for the alloc cache though. This code was not written with threading in mind. I think it could work if you made the creation of the alloc_id atomic with the insertion into the cache. But then you'd be back at troubles with the end position. So if there's an  entry in the alloc_cache but not the end position, you need to decode the allocation, but throw away the result in order to make the deserializer move forward.

Uuuuh. It's a little messy but should be doable. You wanna do a live hacking together to hash out the details or is the above info enoughfor you?